datastore/datastore.go

```diff
@@ -0,0 +1,303 @@
+package datastore
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"time"
+
+	"cloud.google.com/go/datastore"
+	"github.com/GannettDigital/go-newrelic-plugin/helpers"
+	"github.com/Sirupsen/logrus"
+	"golang.org/x/net/context"
+	"golang.org/x/oauth2/google"
+	"google.golang.org/api/monitoring/v3"
+)
+
+var stackdriverEndpoints = []string{
+	"datastore.googleapis.com/api/request_count",
+	"datastore.googleapis.com/index/write_count",
+	//"datastore.googleapis.com/entity/read_sizes", --TODO add distribution data
+	//"datastore.googleapis.com/entity/write_sizes",
+}
+
+//fields for datastore Query
+type DatastoreKind struct {
+	BuiltinIndexBytes   int       `datastore:"builtin_index_bytes"`
+	BuiltinIndexCount   int       `datastore:"builtin_index_count"`
+	CompositeIndexBytes int       `datastore:"composite_index_bytes"`
+	CompositeIndexCount int       `datastore:"composite_index_count"`
+	EntityBytes         int       `datastore:"entity_bytes"`
+	Bytes               int       `datastore:"bytes"`
+	Count               int       `datastore:"count"`
+	KindName            string    `datastore:"kind_name"`
+	Timestamp           time.Time `datastore:"timestamp"`
+}
+
+// PluginData defines the format of the output JSON that plugins will return
+type PluginData struct {
+	Name            string                   `json:"name"`
+	ProtocolVersion string                   `json:"protocol_version"`
+	PluginVersion   string                   `json:"plugin_version"`
+	Metrics         []MetricData             `json:"metrics"`
+	Inventory       map[string]InventoryData `json:"inventory"`
+	Events          []EventData              `json:"events"`
+	Status          string                   `json:"status"`
+}
+
+//fields for stackdriver returns
+type StackdriverMetric struct {
+	TimeSeries []struct {
+		Metric struct {
+			Labels struct {
+				ApiMethod    string `json:"api_method"`
+				ResponseCode string `json:"response_code"`
+			} `json:"labels"`
+			Type string `json:"type"`
+		} `json:"metric"`
+		MetricKind string `json:"metricKind"`
+		Points     []struct {
+			Interval struct {
+				EndTime   time.Time `json:"endTime"`
+				StartTime time.Time `json:"startTime"`
+			} `json:"interval"`
+			Value struct {
+				Int64Value int64 `json:"int64Value,string"`
+			} `json:"value"`
+		} `json:"points"`
+		Resource struct {
+			Labels struct {
+				ModuleID  string `json:"module_id"`
+				ProjectID string `json:"project_id"`
+				VersionID string `json:"version_id"`
+			} `json:"labels"`
+			Type string `json:"type"`
+		} `json:"resource"`
+		ValueType string `json:"valueType"`
+	} `json:"timeSeries"`
+}
+
+type Client struct {
```

Comment:
Standard structure is to include methods that operate on a type right after the type so that you can view them all together.